### DISCUSSION TITLE
lua script runtime error after haproxy update - endless redirect loop

### DISCUSSION THREAD

**Discussion by mamema** [authelia/authelia#5836] [Troubleshooting] [answered]
i've updated haproxy ijn pfsense to (internal version 0.63.1) haproxy is now on haproxy-devel-2.8.d12

since the update i have a endless redirect loop and after digging deep, i've found this haproxy error:

Aug 11 12:21:50 pfSense haproxy[64798]: Lua function 'auth-request': [state-id 0] runtime error: table index is nil from [C]: in method 'get_stats', /var/etc/haproxy/luascript_auth-request.lua:136: in global 'auth_request', /var/etc/haproxy/luascript_auth-request.lua:28: in function line 27.
 
lua scripting is over my head and i need some help here

it is erroring out in this block

-- Check whether the given backend has servers that
	-- are not `DOWN`.
	local addr = nil
	for name, server in pairs(core.backends[be].servers) do
		local status = server:get_stats()['status']
		if status == "no check" or status:find("UP") == 1 then
			addr = server:get_addr()
		   break
		end
	end

**Comment by james-d-elliott** [comment]
After taking a closer look / investigation it looks like it's most likely a HAProxy 2.8.0 bug which they fixed in 2.8.1: https://github.com/TimWolla/haproxy-auth-request/issues/58

**Reply by mamema** [reply]
uh, thank you. This is a bummer, cause you've discussed this in June already, but i've update haproxy via the offical pfsense gui on 9th of august to haproxy 2.8.
So till this bug gets fixed in haproxy on pfsense, this will be november.....
Are there possible (comment out?) workarounds in the lua script to get authelia running again?

**Reply by james-d-elliott** [reply, marked as answer]
HAProxy doesn't properly run the lua script, so it's quite out of our control. The bug was fixed in the latest release of HAProxy itself, so your options are probably to figure out if Netgate can provide a patch earlier as the upgrade to 2.8 without the patch was a breaking change or downgrade if that's possible.

The issue is the lua function to get the response code (which is absolutely required for it to function) fails with an error because of a bug in HAProxy.

**Reply by mamema** [reply]
i've found a workaround. As the haproxy-devel package was updated and non working (haproxy-devel 2.8ish) removing this package and reinstalling the haproxy NON-devel version, which is on 2.7ish fixed the authelia endless loop issue

**Reply by aurrak** [reply]
Just ran into the same issue. Thank you @mamema for the workaround!

How could you tell that the next haproxy-devel update in pfsense will be around November? Is that published somewhere?

**Reply by mamema** [reply]
just a gut feeling as long time pfsense user. November seems valid only the year is unclear. :-)
